### PR TITLE
Use gogo Protobuf generator for custom types in Device Protobuf

### PR DESCRIPTION
**Pull request by kuujo**
This PR changes the Protobuf code generator to [gogo](https://github.com/gogo/protobuf). This allows for much more control over the Golang code generation. The resulting `Device` type now supports a custom `ID` and `Version` type, uses conventional names for acronym fields, and generates code for encoding/decoding messages without reflection.

### CODE REVIEW

File: pkg/northbound/device/device.proto [hunk 1]
```diff
@@ -41,15 +42,15 @@ message UpdateRequest {
 
 // UpdateResponse is sent in response to an UpdateDeviceRequest
 message UpdateResponse {
-    // metadata is the updated device metadata
-    ObjectMetadata metadata = 1;
+    // version is the updated version of the device
+    uint64 version = 1 [(gogoproto.casttype) = "Version"];
```
Comment (SeanCondon) [inline pkg/northbound/device/device.proto]: is this semantic versioning e.g. 1.0.0 serialized in to one number?
Super magic powers going on here - I wonder how non go code (e.g. Typescript) would deserialize it?

File: pkg/northbound/device/device.proto [hunk 2]
```diff
@@ -16,10 +16,11 @@ limitations under the License.
 
 syntax = "proto3";
 
-// Package admin defines the administrative and diagnostic gRPC interfaces.
+// Package topo.device defines interfaces for managing devices.
 package topo.device;
```
Comment (SeanCondon) [inline pkg/northbound/device/device.proto]: nice

File: pkg/northbound/device/device.proto [hunk 3]
```diff
@@ -125,13 +126,13 @@ message Device {
     string software_version = 5;
```
Comment (SeanCondon) [inline pkg/northbound/device/device.proto]: is this related to version above?

File: pkg/northbound/device/device.proto (hunk 1 above)
Comment (kuujo) [reply pkg/northbound/device/device.proto]: This is a revision version assigned by the store. It's used to prevent two concurrent changes from being made to the same version of the same object. This is the pattern the Kubernetes API uses to provide concurrency control as well. Basically, when an object is updated, the object is returned with a version number. Then, if the object is modified again, the modified object's version number is compared to the store to determine whether an earlier change to the same version was already made, and rejects the change if so, forcing the client to read the updated object state and retry. We can also rename this field `Revision` to avoid confusion.

The custom `Version` type is only relevant to Golang code generation. The `casttype` annotation just casts the `uint64` field to `Version` when it's read from the wire, and casts `Version` back to `uint64` when it's written. In other languages, the `gogoproto.casttype` annotation is unknown, so they will generate code that just operates on a `uint64` type.

File: pkg/northbound/device/device.proto (hunk 3 above)
Comment (kuujo) [reply pkg/northbound/device/device.proto]: This can become `version`

File: pkg/northbound/device/device.proto (hunk 1 above)
Comment (SeanCondon) [reply pkg/northbound/device/device.proto]: ok - thanks for the explaination
